plenum/server/consensus/ordering_service.py

```diff
@@ -1,9 +1,173 @@
+from plenum.common.util import compare_3PC_keys
 from plenum.server.consensus.consensus_data_provider import ConsensusDataProvider
+from plenum.server.replica_validator_enums import INCORRECT_INSTANCE, DISCARD, INCORRECT_PP_SEQ_NO, ALREADY_ORDERED, \
+    STASH_VIEW, FUTURE_VIEW, OLD_VIEW, PROCESS, GREATER_PREP_CERT, STASH_CATCH_UP, CATCHING_UP, OUTSIDE_WATERMARKS, \
+    STASH_WATERMARKS
 
 
 class OrderingService:
     def __init__(self, data: ConsensusDataProvider, bus: InternalBus, network: ExternalBus):
         self._data = data
         self._bus = bus
         self._network = network
+        self._stasher = StashingRouter()
+
+        self._stasher.subscribe(PrePrepare, self.process_preprepare)
+        self._stasher.subscribe(Prepare, self.process_prepare)
+        self._stasher.subscribe(Commit, self.process_commit)
+        self._stasher.subscribe_to(network)
+
+        # ToDo: need to define how to get current value (when object instantiating) for next statuses vars
+        self._vc_in_progress = False
+        self._is_participating = False
+        self._is_synced = False
+
+        self._bus.subscribe(ViewChangeInProgress, self.set_vc_in_progress)
+        self._bus.subscribe(LedgerSyncStatus, self.set_is_synced)
+        self._bus.subscribe(ParticipatingStatus, self.set_is_participating)
+
+    def process_preprepare(self, pre_prepare: PrePrepare, sender: str):
+        result, reason = self._validate(pre_prepare)
+        if result == PROCESS:
+            self._process_preprepare(pre_prepare, sender)
+        return result
+
+    def process_prepare(self, prepare: Prepare, sender: str):
+        result, reason = self._validate(prepare)
+        if result == PROCESS:
+            self._process_prepare(prepare, sender)
+        return result
+
+    def process_commit(self, commit: Commit, sender: str):
+        result, reason = self._validate(commit)
+        if result == PROCESS:
+            self._process_commit(commit, sender)
+        return result
+
+    def _process_preprepare(self, pre_prepare: PrePrepare, sender: str):
+        pass
+
+    def _process_prepare(self, prepare: Prepare, sender: str):
+        pass
+
+    def _process_commit(self, commit: Commit, sender: str):
+        pass
+
+    @property
+    def view_no(self):
+        return self._data.view_no
+
+    @property
+    def legacy_last_prepared_sertificate(self):
+        """
+        We assume, that prepared list is an ordered list, and the last element is
+        the last quorumed Prepared
+        """
+        if self._data.prepared:
+            last_prepared = self._data.prepared[-1]
+            return last_prepared.view_no, last_prepared.pp_seq_no
+        return self.last_ordered_3pc
+
+    @property
+    def vc_in_progress(self):
+        return self._vc_in_progress
+
+    @property
+    def is_synced(self):
+        return self._is_synced
+
+    @property
+    def is_participating(self):
+        return self._is_participating
```

Comment:
I think this one probably needs to be located in `self._data.is_participating`

Reply:
ok